TravelAgency-spring-mvc/src/main/java/cz/muni/fi/travelAgency/controllers/CustomerController.java

```diff
@@ -72,21 +79,25 @@ public String getEdit(@PathVariable long id, Model model) {
         return "customer/edit";
     }
 
-    @RequestMapping(value = "/editCustomer", method = RequestMethod.POST)
+    @RequestMapping(value = "editCustomer", method = RequestMethod.POST)
     public String submitEdit(@Valid @ModelAttribute("customer") CustomerCreateDTO createDTO,
                              BindingResult bindingResult, Model model, RedirectAttributes redirectAttributes,
                              UriComponentsBuilder uriBuilder) {
-        if (!resultIsValid(bindingResult, model)) {
+//        if (!resultIsValid(bindingResult, model)) {
+//            return "customer/edit";
+//        }
+        if (createDTO == null)
+        {
             return "customer/edit";
         }
         CustomerDTO customerDTO = mapper.mapTo(createDTO, CustomerDTO.class);
         customerDTO.setPasswordHash(createDTO.getPassword());
         customerFacade.updateCustomer(customerDTO);
         redirectAttributes.addFlashAttribute("alert_success", "Customer" + createDTO.getSurname() + " updated");
-        return "redirect:" + uriBuilder.path("customer/edit" + createDTO.getId()).build().encode().toUriString();
+        return "redirect:" + uriBuilder.path("/admin/customer/detail/" + createDTO.getId()).build().encode().toUriString();
```

Comment:
This is a tricky one. You are sending any user that edited his account to a page he will not have access to. I suggest moving the admin/customer/detail to the customers scope (the page and rather adjust admin controllers to access the page customer/detail)